recipe/meta.yaml

```diff
@@ -24,72 +22,73 @@ source:
-      -DBUILD_META_EXAMPLES=OFF              {{ lc }}
-      -DBUNDLE_JSON=OFF                      {{ lc }}
-      -DBUNDLE_NLOPT=OFF                     {{ lc }}
-      -DUSE_SVMLIGHT=OFF                     {{ lc }}
-      -DENABLE_TESTING=OFF                   {{ lc }}
-      -DENABLE_COVERAGE=OFF                  {{ lc }}
-      -DUSE_SWIG_DIRECTORS=ON
-    if errorlevel 1 exit 1  # [win]
-
+    cmake ..                                                  {{ lc }}
+      -G"{{vs}}CMAKE_GENERATOR{{ve}}"                         {{ lc }}
+      -DCMAKE_PREFIX_PATH="{{ prefix }}"                      {{ lc }}
+      -DCMAKE_INSTALL_PREFIX="{{ prefix }}"                   {{ lc }}
+      -DCMAKE_BUILD_TYPE=Release                              {{ lc }}
+      -DBUILD_EXAMPLES=OFF                                    {{ lc }}
+      -DBUILD_META_EXAMPLES=OFF                               {{ lc }}
+      -DBUNDLE_JSON=OFF                                       {{ lc }}
+      -DBUNDLE_NLOPT=OFF                                      {{ lc }}
+      -DUSE_SVMLIGHT=OFF                                      {{ lc }}
+      -DENABLE_TESTING=OFF                                    {{ lc }}
+      -DENABLE_COVERAGE=OFF                                   {{ lc }}
+      -DUSE_SWIG_DIRECTORS=ON                                 {{ lc }}
+      -DLICENSE_GPL_SHOGUN=OFF                                {{ lc }}  # [blas == 'mkl']
+      -DUSE_SVMLIGHT=OFF                                      {{ lc }}  # [blas == 'mkl']
+      -DBLA_VENDOR={{ bla_vendor }}                           {{ lc }}
+      -DLAPACK_LIBRARIES={{ prefix }}/lib/{{ blas_lib }}      {{ lc }}
+      -DBLAS_LIBRARIES={{ prefix }}/lib/{{ blas_lib }}
     cmake --build . --target shogun --config Release -- {{ parbuild }}
-    if errorlevel 1 exit 1  # [win]
-
     cmake --build . --target install --config Release -- {{ parbuild }}
-    if errorlevel 1 exit 1  # [win]
-
 requirements:
   build:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
     - python
-    - cmake
+    - cmake =3.15
   host:
-    - arpack       # [not win]
+    - arpack
     - bzip2
     - ctags
-    - curl
+    - libcurl
+    - libgomp       # [linux]
     - eigen
     - glpk
     - hdf5
-    - json-c       # [not win]
+    - json-c =0.12
     - libxml2
     - lzo
-    - openblas     # [not win]
-    - libprotobuf  # [not win]  # conda-forge doesn't have shared lib yet
-    - rxcpp 4.0.0
+    - mkl-devel     # [blas == 'mkl']
+    - openblas      # [blas == 'openblas']
+    - libprotobuf
+    - rxcpp 4.1.0
     - snappy
     - xz
     - zlib
   run:
-    - arpack       # [not win]
+    - arpack
     - bzip2
     - ctags
-    - curl
+    - libcurl
+    - libgomp       # [linux]
     - glpk
     - hdf5
-    - json-c       # [not win]
+    - json-c =0.12
     - libxml2
     - lzo
-    - openblas     # [not win]
-    - libprotobuf  # [not win]
+    - blas * {{ blas }}
+    - libopenblas   # [blas == 'openblas']
```

Comment:
This should be done as follows: https://conda-forge.org/docs/maintainer/knowledge_base.html#blas

Reply:
yeah i actually saw these. the only thing is that during cmake/compile time there are some lapack(e) checks that enables further optimizations within the code, i.e. some of the methods are significantly faster...
see the case of mkl here: https://github.com/shogun-toolbox/shogun/blob/develop/cmake/ShogunFindLAPACK.cmake#L41-L83

Reply:
i mean if this is the way then this is the way... happy to change it to get it merged. just saying the context why i created it like this

Reply:
Sure, so you can restrict it to only using MKL or openblacs, but it has to be done in the way listed in the docs there

Reply:
@scopatz lemme know what you think... i would really get this merged as soon as.

Reply:
i see, but then these compile time decisions will not work :(
i'll change the PR and see if i get it right at first shot

Reply:
@scopatz just one last question: imo one cannot use MKL with gpl library. is that true or i misunderstood this, see: https://github.com/conda-forge/shogun-cpp-feedstock/pull/34/files#diff-e178b687b10a71a3348107ae3154e44cR46